src/Altinn.App.Core/Internal/Expressions/LayoutEvaluator.cs

```diff
@@ -99,18 +99,19 @@ private static void RunLayoutValidationsForRequiredRecurs(List<ValidationIssue>
                 {
                     if (state.GetModelData(binding, context) is null)
                     {
+                        var field = state.AddInidicies(binding, context);
                         validationIssues.Add(new ValidationIssue()
                         {
                             Severity = ValidationIssueSeverity.Error,
                             InstanceId = state.GetInstanceContext("instanceId").ToString(),
                             DataElementId = dataElementId,
-                            Field = state.AddInidicies(binding, context),
-                            Description = "TODO required",
+                            Field = field,
+                            Description = $"{field} is required in component with id {context.Component.Id}",
```

Comment:
If I remember correctly we use a text resource id in the description field in order to be able to localize the error messages.

Reply:
Yes, but I hope that in the future we’ll use `code` for that purpose. (At least if it says `required`)

see https://github.com/Altinn/app-frontend-react/issues/1062